### PR TITLE
test/cls_rgw_gc: verify that `cls_rgw_gc_queue_list_entries()` returns `max-1` entries

### PR DESCRIPTION
**Pull request by AmnonHanuhov**
CC: @ofriedma 

Signed-off-by: Amnon Hanuhov <ahanukov@redhat.com>

### CODE REVIEW

File: src/test/cls_rgw_gc/test_cls_rgw_gc.cc [hunk 1]
```diff
@@ -260,22 +260,14 @@ TEST(cls_rgw_gc, gc_queue_ops3)
...
-      ASSERT_EQ("chain-1", it.tag);
-    }
-    if (i == 1) {
-      ASSERT_EQ("chain-0", it.tag);
-    }
-    i++;
-  }
+  auto it = list_info1.cbegin();
+  std::cerr << "[          ] list info tag = " << it->tag << std::endl;
+  ASSERT_EQ("chain-1", it->tag);
```
Comment (tchaikov) [inline src/test/cls_rgw_gc/test_cls_rgw_gc.cc:267]: @pritha-srivastava how is this exercised? after looking around, i found that it was neither registered as a unit nor is used by any qa suite.

File: src/test/cls_rgw_gc/test_cls_rgw_gc.cc [hunk 2]
```diff
@@ -260,22 +260,14 @@ TEST(cls_rgw_gc, gc_queue_ops3)
...
-    if (i == 0) {
-      ASSERT_EQ("chain-1", it.tag);
-    }
-    if (i == 1) {
-      ASSERT_EQ("chain-0", it.tag);
-    }
-    i++;
-  }
+  auto it = list_info1.cbegin();
+  std::cerr << "[          ] list info tag = " << it->tag << std::endl;
```
Comment (tchaikov) [inline src/test/cls_rgw_gc/test_cls_rgw_gc.cc:266]: no need to print out `it->tag`. `ASSERT_EQ()` does this for us if the lhs and rhs are not equal.

**Review by tchaikov — CHANGES_REQUESTED** [review]
the title of the commit message reads like a bug report. ideally, the title of the commit message should use the imperative mood. see also https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes . please use `git commit --amend`  and `git push -f` for revising the commit message and its title.

better off revising the title like:

> test/cls_rgw_gc: verify that cls_rgw_gc_queue_list_entries() returns "max - 1" entries

File: src/test/cls_rgw_gc/test_cls_rgw_gc.cc (hunk 2 above)
Comment (AmnonHanuhov) [reply src/test/cls_rgw_gc/test_cls_rgw_gc.cc]: thanks, fixed and repushed.

**Comment by mattbenjamin** [comment]
unstahl

**Pull request closed without merging**